lib/ffi.dart

```diff
@@ -2,21 +2,32 @@
 // generated by flutter_rust_bridge_codegen.
 
 import 'dart:ffi';
+import 'dart:io' as io;
 
 import 'bridge_generated.dart';
 import 'bridge_definitions.dart';
-export 'bridge_definitions.dart';
 
 // Re-export the bridge so it is only necessary to import this file.
+export 'bridge_definitions.dart';
 export 'bridge_generated.dart';
-import 'dart:io' as io;
 
 const _base = 'native';
+// Use for Darwin framework lookup under the DEBUG mode.
+// The value must be equal to the package name defined in pubspec.yaml.
+const _package = 'native';
 
 // On MacOS, the dynamic library is not bundled with the binary,
 // but rather directly **linked** against the binary.
 final _dylib = io.Platform.isWindows ? '$_base.dll' : 'lib$_base.so';
 
-final Native api = NativeImpl(io.Platform.isIOS || io.Platform.isMacOS
-    ? DynamicLibrary.executable()
-    : DynamicLibrary.open(_dylib));
+final Native api = NativeImpl(
+  io.Platform.isIOS || io.Platform.isMacOS
+      ? _kDebugMode
+          ? DynamicLibrary.open('$_package.framework/$_package')
+          : DynamicLibrary.executable()
+      : DynamicLibrary.open(_dylib),
+);
```

Comment:
I checked how I implement the v2 loader - from the comments, one reference is https://github.com/flutter/flutter/blob/8b6277e63868c2029f1e2327879b7899be44fbe2/packages/flutter_tools/templates/plugin_ffi/lib/projectName.dart.tmpl#L47-L58.

It seems that, under both debug and release mode, the symbols are from `'$_libName.framework/$_libName'`. Therefore, maybe even in release mode we should use `open`?

![image](https://github.com/Desdaemon/flutter_rust_bridge_template/assets/5236035/a5006e03-4094-4c46-8c3d-c999cffb5ffb)

Reply:
Interesting. I'm not actively using Flutter on MacOS right now so I'll wait for more reports to stream in if it works or not.

Reply:
I can also confirm that, using `flutter build macos` on frb_example, I generated an app and I can open and run it well on macos. It is based on v2's default loader logic, thus is using the `open` on debug and release.

Reply:
> I checked how I implement the v2 loader - from the comments, one reference is https://github.com/flutter/flutter/blob/8b6277e63868c2029f1e2327879b7899be44fbe2/packages/flutter_tools/templates/plugin_ffi/lib/projectName.dart.tmpl#L47-L58.

I guess we should follow this then. I'll run another test for releases.